buildSrc/src/main/java/org/elasticsearch/gradle/DistributionDownloadPlugin.java

```diff
@@ -187,25 +201,31 @@ private static void setupDownloadServiceRepo(Project project) {
      */
     private Object dependencyNotation(Project project, ElasticsearchDistribution distribution) {
 
-        if (Version.fromString(VersionProperties.getElasticsearch()).equals(distribution.getVersion())) {
-            return projectDependency(project, distributionProjectPath(distribution), "default");
-            // TODO: snapshot dep when not in ES repo
-        }
-        BwcVersions.UnreleasedVersionInfo unreleasedInfo = bwcVersions.unreleasedInfo(distribution.getVersion());
-        if (unreleasedInfo != null) {
-            assert distribution.getBundledJdk();
-            return projectDependency(project, unreleasedInfo.gradleProjectPath, distributionProjectName(distribution));
+        if (ClasspathUtils.isElasticsearchProject(project)) {
+            // non-external project, so depend on local build
+
+            if (VersionProperties.getElasticsearch().equals(distribution.getVersion())) {
+                return projectDependency(project, distributionProjectPath(distribution), "default");
+                // TODO: snapshot dep when not in ES repo
+            }
+            BwcVersions.UnreleasedVersionInfo unreleasedInfo = bwcVersions.unreleasedInfo(Version.fromString(distribution.getVersion()));
+            if (unreleasedInfo != null) {
+                assert distribution.getBundledJdk();
+                return projectDependency(project, unreleasedInfo.gradleProjectPath, distributionProjectName(distribution));
+            }
         }
 
         if (distribution.getType() == Type.INTEG_TEST_ZIP) {
             return "org.elasticsearch.distribution.integ-test-zip:elasticsearch:" + distribution.getVersion();
         }
 
+
+        Version distroVersion = Version.fromString(distribution.getVersion());
```

Comment:
What was the motivation of changing `ElasticsearchDistribution` to store the version as a `String` if that then forces us to re-parse it into a `Version` in places. Was calling `toString()` too cumbersome?

Reply:
No, we need it to be a String or we lose -SNAPSHOT. Version ignores this, thus 8.0.0-SNAPSHOT becomes 8.0.0.

Reply:
Ah I see. Should we make `Version` so that it doesn't drop these qualifiers? The parser pattern seems to explicitly support them, we just don't keep them around.

Reply:
Version matches up with the class of the same name within the server module which represents a version in code. This does not have the concept of snapshot (or alpha/beta/rc anymore). The String here represents the version that we could have in a released artifact, which can have those qualifiers. We use Version when we want to compare the numeric portions with conditional logic. So I think the current behavior is what we want.